build-logic/src/main/kotlin/ktlint-kotlin-common.gradle.kts

```diff
@@ -5,22 +5,33 @@ plugins {
     kotlin("jvm")
 }
 
+val projectLibs = extensions.getByType<VersionCatalogsExtension>().named("libs")
```

Comment:
`ktlint-kotlin-common` is a Gradle plugin, so it can't use the synthetic `libs.versions.java.compilation` accessor. I had to extract version the _native_ way via Gradle extensions